app/assets/javascripts/features/auto-search.js

```diff
@@ -0,0 +1,143 @@
+$(function(){
+  var search_sidebar_template = JST['templates/search_sidebar'],
+
+      search_project_template = JST['templates/search/project'],
+      search_story_template = JST['templates/search/story'],
+      search_task_template = JST['templates/search/task'],
+      search_comment_template = JST['templates/search/comment'],
+      search_attachment_template = JST['templates/search/attachment'],
+
+      search_animation_speed = 500,
+      search_timeout = false;
+
+  // clicking outside the search sidebar closes it
+  $(document).on("click", function() {
+    closeSearchSidebar();
+  });
+  $(document).on("click", "#search-sidebar", function(ev) {
+    ev.stopPropagation();
+  });
+
+  // open the sidebar
+  $(document).on('click', '.ui-dropdown-menu.search', function(ev){
+    openSearchSidebar();
+  });
+  $(document).on('search:open', function(){
+    openSearchSidebar();
+  });
+
+  // pressing esc key
+  $(document).on('dialogs:close', function(){
+    closeSearchSidebar();
+  });
+
+  // do the search when typing in the search field
+  $(document).on('keyup', '#search-sidebar input[name=q]', function(ev){
+    // esc key
+    if (ev.keyCode == 27) {
+      closeSearchSidebar();
+
+    } else {
+      $('#search-sidebar').addClass('searching');
+      $('#search-sidebar .no-results').hide();
+      $('#search-sidebar .search-results').hide();
+      $('#search-sidebar .search-results ul').html('');
+
+      clearTimeout(search_timeout);
+      search_timeout = setTimeout(function() {
+        performSearch();
+      }, 500);
+    }
+  });
+
+  // do not actually submit the form
+  $(document).on('submit', '#search-sidebar form', function(ev){
+    ev.preventDefault();
+  });
+
+  function openSearchSidebar(){
+    $('body').addClass('search-sidebar-open');
+    $('body').append(search_sidebar_template());
+    $('#search-sidebar input').focus();
+    $('#search-sidebar').animate({right: 0}, search_animation_speed);
```

Comment:
we should do the animation in CSS by adding an open class to it. Means we can responsively animate, if we didn’t want the animation to do the same thing on mobile